### ISSUE TITLE
[no-type-alias] false positive

### ISSUE THREAD

**Issue by artola** [typescript-eslint/typescript-eslint#3321]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-type-alias": ["error", {"allowAliases": "always"}]
  }
}
```

```TS
type Foo = typeof import('foo');
```

**Expected Result**

**Actual Result**

```txt
[@typescript-eslint/no-type-alias] Type unhandled are not allowed.ts(30010)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.22.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.24.0` |
| `node`                             | `12.19.0` |

**Comment by bradzacher** [comment]
why not just `import type Foo from 'foo'`?

**Comment by artola** [comment]
I use it to workaround in `foo.d.ts` as here:

```
type Foo = typeof import('foo');

declare module 'foo/setup' {
  export default function setup(options?: {
     ...
  }): Foo;
}
```

**Comment by bradzacher** [comment]
you can move the import inside the `declare`

```diff
- type Foo = typeof import('foo');

  declare module 'foo/setup' {
+   import type Foo from 'foo'?
    export default function setup(options?: {
       ...
    }): Foo;
  }
```

**Comment by artola** [comment]
@bradzacher this would make it accesible just inside the `declare`, while before it also was exposed.

Might be that someone points out a better way to achieve what I need, and then the rule will not kick-in.
But, I opened this issue because anyway the rule fails with the options `"allowAliases": "always"` and should not.
